### ISSUE TITLE
Spanish Version

### ISSUE THREAD

**Issue by bastislack** [bastislack/highline-freestyle#160]
There was some feedback that a spanish version would be much appreciated

**Comment by bastislack** [comment]
@rorystephenson Do you have experience on how to best implement other languages in an app?

**Comment by rorystephenson** [comment]
@bastislack Yep I have some experiences from other programming languages/frameworks. The general idea is to extract all text which needs to be translated out of the code and in to dedicated language files and then have a reference to them in the code via a key.

In practice it's worth using a dedicated package for this as they can save you a lot of time implementing your own solution and sometimes offer more advanced features. I'm going to have a look at what packages are popular for React.

**Comment by rorystephenson** [comment]
@bastislack I've had a go at an implementation, have a look at #173 .

Right now the only things that are translated are:

- Page names
- Language names
- The word "level" on the trickDetails page.

I think it would be better to go with a separate PR to add all the other translations to avoid a massive conflict mess.

Let me know what you think!

**Comment by bastislack** [comment]
That's super cool, I just had a look at the code. I will checkout the PR later and have a look at it.

**Comment by bastislack** [comment]
I played around a bit with it and when I scroll down on the trick or combo list and then tap on one of the items it switches back from spanish to english. It also happens when I try to generate a combo.

**Comment by rorystephenson** [comment]
@bastislack Good spotting that's quite strange 🤔 . I'll have a look.

**Comment by rorystephenson** [comment]
So I separated out localisation from App.js to LocalizedApp.js to tidy things up a bit and that actually _seems_ to have solved the issue for me locally. Maybe if it's playing up for you try closing the tab and opening a new one. Let me know how it goes.

---

While I was at it I added a webpack loader so we don't have to manually run the compile command when changing translations.

**Comment by bastislack** [comment]
Seems to work fine now!
I would wait with merging it to the develop branch until we finished the next milestone.
Then we can focus on the translation to spanish :)
